### ISSUE TITLE
[configurator] New quoting rules may create problems.

### ISSUE THREAD

**Issue by ejgallego** [ocaml/dune#1833]
New quoting rules in 1.7.0 means that:
```
module C = Configurator.V1
let _ = C.Pkg_config.query p ~package:"gtk+-3.0 >= 3.10" 
```
will always return `None` as the package name is quoted in such a way that `pkg-config` will get confused.

This can be worked around by using:
```
let _ = C.Pkg_config.query p ~package:"gtk+-3.0>=3.10" 
```
however it is not clear to me that the quoting in 1.7.0 is correct.

**Comment by ejgallego** [comment]
This may be more serious than what I though, as the below solution seems broken on OSX:
```
-> stderr:
run: /usr/local/bin/pkg-config gtkspell3-3.0>=3.0.4
Fatal error: exception Sys_error("/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/ocaml-configuratore0f15a/stdout-2: No such file or directory")
```

**Comment by rgrinberg** [comment]
What new quoting rules are you talking about? I can't recall any changes here.

**Comment by ejgallego** [comment]
Let me locate the commit on `src/configurator` but indeed it seems that maybe now some commands are being quoted twice.

**Comment by ejgallego** [comment]
The bisect was a pain for other reasons but indeed seems that a5becc4e68bc129b520c59f0228f4c31cba2499b creates this problem.

**Comment by ejgallego** [comment]
Incorrect quoting on Linux is:
```
run: /usr/bin/pkg-config ''\''gtk+-3.0 >= 3.18'\'''
```

**Comment by rgrinberg** [comment]
Okay, got it. @Chris00 do you mind restoring the old quoting behavior?

**Comment by diml** [comment]
I think that just removing the `quote_if_needed` calls in `Pkg_config.query` should be enough. The quoting is now handled by `Process.run`, so things are double quoted at the moment.

**Comment by Chris00** [comment]
I'll try to take a look to that today.

**Comment by rgrinberg** [comment]
#1834 is my attempt to address this

**Comment by Chris00** [comment]
@rgrinberg Sorry, I just saw you attempt.  (I tested and pushed a fix that is quite similar to yours.)

**Comment by Chris00** [comment]
> The bisect was a pain for other reasons

@ejgallego May you explain a bit more why?

**Comment by Chris00** [comment]
@rgrinberg I'll also submit an improved doc for `query`—I think explaining the possibility to put a constraint in the query would be useful for our users.

**Comment by ejgallego** [comment]
> @ejgallego May you explain a bit more why?

Sure, I just did the bisect on a live switch and I got some problems with `make install` not working, also some trouble due to `dune-package`.

**Comment by ejgallego** [comment]
Quoting seems still broken on OSX, will investigate more [no OSX machine at hand]

**Comment by rgrinberg** [comment]
Indeed I see the issue - It's related to our janky homebrew support. Fixing.

**Comment by ejgallego** [comment]
TTBOMK closed by #1842
